packages/code-analyzer-core/src/messages.ts

```
@@ -21,20 +59,29 @@ const MESSAGE_CATALOG : MessageCatalog = {
     PluginErrorFromGetAvailableEngineNames:
         `Failed to add engine plugin. The plugin's getAvailableNames method threw an error:\n%s`,
 
-    PluginErrorFromCreateEngine:
-        `Failed to create engine with name '%s'. The plugin's createEngine method threw the following error:\n` +
+    PluginErrorWhenCreatingEngine:
+        `Failed to create engine with name '%s' due to the following error:\n` +
         `%s\n\n` +
         `If you wish to ignore this error and disable this engine, then update your Code Analyzer configuration with:\n` +
         `engines:\n` +
         `  %s:\n` +
         `    disable_engine: true\n`,
 
+    PluginErrorWhenCreatingEnginePriorToDisableEngineCheck:
+        `Failed to create engine with name '%s' due to the following error:\n%s\n`,
```

It feels a little odd for these to be separate messages. Maybe you could have "Failed to create..." be one message and "If you wish to ignore this error and disable this engine..." be a second message, and then you append the second to the first in situations where it's relevant?

---

Good idea.